validator/src/main/java/com/google/daq/mqtt/registrar/Registrar.java

```diff
@@ -508,13 +509,24 @@ private void shutdown() {
     }
   }
 
-  private Map<String, LocalDevice> loadLocalDevices() {
+  private Map<String, LocalDevice> loadLocalDevices(Set<String> specifiedDevices) {
     Preconditions.checkNotNull(siteDir, "site directory");
     File devicesDir = new File(siteDir, DEVICES_DIR);
     if (!devicesDir.isDirectory()) {
       throw new RuntimeException("Not a valid directory: " + devicesDir.getAbsolutePath());
     }
-    String[] devices = devicesDir.list();
+
+    String[] devices;
+    if (specifiedDevices == null) {
+      devices = devicesDir.list();
+    } else {
+      devices = devicesDir.list(new FilenameFilter() {
+      public boolean accept(File dir, String name) {
```

Comment:
something seems off about the indent... the "public" should not line up with "devices"  -- the lint checks are complaining because of "  Checkstyle violations by severity: [warning:151]" -- and I think this is because they are capped at 150 (previous high-water mark).  You should be able to look through the logs there to see exactly what the errors are (search for the filename). Ideally you'd also see these in the IDE or running "validator/bin/build check" locally on the command line.

Reply:
Fixed this which should -1 lint problem also improve -1 more lint by moving null overload adjacent